### PR TITLE
Tested pizza hut application

### PR DESCRIPTION
**Pull request by dhruvi-kamani**
Created POC on cypress-typescript while testing pizza hut application.

### CODE REVIEW

File: cypress/e2e/test.spec.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+				orderPizzaPage.pizzaNameInCard
+					.then($element => {
+						cy.wrap($element.contents()[0].textContent).as("pizzaName")
+					});
+				orderPizzaPage.pizzaPriceInCard
+					.invoke("text")
+					.as("pizzaPrice");
+				orderPizzaPage.pizzaAddButtonInCard
+					.click();
+			});
```
Comment (ajinkyaces) [inline cypress/e2e/test.spec.ts]: Not clear what is happening here. Please explain

File: cypress/e2e/test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		modifyOrderPage.addToppingButton.eq(0).click();
+		modifyOrderPage.modifiedPrice.then(($span) => {
+			cy.wrap($span.contents()[1].textContent).as("pizzaPrice");
+		});
+		modifyOrderPage.addToBasketButton.click();
+
+		cy.get("@pizzaPrice").then((price) => {
+			basketPage.subTotalValue
+				.should("contain.text", price);
+		});
```
Comment (ajinkyaces) [inline cypress/e2e/test.spec.ts]: Please explain and am guessing you are removing first product found can we keep specific data which needs to be removed.

File: cypress/utility.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const dataSynth = (selector: string) => `[data-synth="${selector}"]`;
+export const partialDataSynth = (selector: string) => `[data-synth*=${selector}]`;
```
Comment (ajinkyaces) [inline cypress/utility.ts:2]: Good approach, but the locator is still hard coded from respective functions. Can we have common place for storing locators ?

File: cypress/support/e2e.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	cy.visit("/");
+	cy.waitForApi("@loadPageRequests", 8);
+
+	cy.ifExist(dataSynth("button--delocalise"), ($element) => {
+		$element.click();
+	});
+
+	homePage.collectFromStoreButton.should("be.visible").click();
+	homePage.searchForLocation("Wakad");
+	storePage.orderButton.should("have.text", "Order");
```
Comment (ajinkyaces) [inline cypress/support/e2e.ts]: Location should be stored in fixture

File: cypress/support/commands.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+	return cy.wrap(values);
+});
+
+Cypress.Commands.add("ifExist", (selector, callback) => {
+	let length = Cypress.$(selector).length;
+	if (length) {
+		callback(cy.get(selector));
+	}
+});
```
Comment (ajinkyaces) [inline cypress/support/commands.ts:64]: Do we need this command ? Its just verifying the presence of element can we use cypress inbuild function ?

**Review by ajinkyaces — CHANGES_REQUESTED** [review]

File: cypress/e2e/test.spec.ts (hunk 1 above)
Comment (dhruvi-kamani) [reply cypress/e2e/test.spec.ts]: After selecting the targeted pizza I am storing PizzaName and PizzaPrice as Alias. And then clicking add button so I can verify those two latter.

File: cypress/e2e/test.spec.ts (hunk 2 above)
Comment (dhruvi-kamani) [reply cypress/e2e/test.spec.ts]: I am not using testIsolation therefore it is continuation of previous test. So, I am removing first one available. Also, there is a possibility that pizza I am targeting may not be available at certain time. So, it is a generic scenario.

Comment (dhruvi-kamani) [reply cypress/e2e/test.spec.ts]: Also, if you are wondering about `$span.contents()[1].textContent` then the text I want to find is in place without any element wrapping it. therefore, I had to do it to extract just text without any side tags or something.

File: cypress/utility.ts (hunk 3 above)
Comment (dhruvi-kamani) [reply cypress/utility.ts:2]: I am following POM so as far as I know all the locators I have placed in respective pages. Please clarify it if you meant something else entirely.

File: cypress/support/commands.ts (hunk 5 above)
Comment (dhruvi-kamani) [reply cypress/support/commands.ts:64]: In the site there is one modal which randomly appears. So in order to close it I need this command. 
with cy.get() it thorws error i.e element not found.

File: cypress/support/e2e.ts (hunk 4 above)
Comment (dhruvi-kamani) [reply cypress/support/e2e.ts]: Done

File: cypress/e2e/test.spec.ts (hunk 1 above)
Comment (ajinkyaces) [reply cypress/e2e/test.spec.ts]: Would appreciate if added few comments for steps or use reporting to address these steps

Comment (dhruvi-kamani) [reply cypress/e2e/test.spec.ts]: Done

**Review by ajinkyaces — APPROVED** [review]

**Pull request merged**